benchmarks/dali_vs_python/BM_jasper/model_repository/jasper_dali/pipeline.py

```diff
@@ -0,0 +1,192 @@
+                device="cpu", range=resample_range
+            )
+
+    if train_pipeline and speed_perturbation_coeffs is not None:
+        dec_sample_rate_arg = speed_perturbation_coeffs * sample_rate
+    elif resample_range is None:
+        dec_sample_rate_arg = sample_rate
+    else:
+        dec_sample_rate_arg = None
+
+    audio, _ = fn.decoders.audio(
+        encoded, sample_rate=dec_sample_rate_arg, dtype=types.FLOAT, downmix=True
+    )
+    if do_remove_silence:
+        begin, length = fn.nonsilent_region(audio, cutoff_db=silence_threshold)
+        audio = fn.slice(audio, begin, length, axes=[0])
+
+    # Max duration drop is performed at DataLayer stage
+
+    if preprocessing_device == "gpu":
+        audio = audio.gpu()
+
+    if dither_coeff != 0.0:
+        audio = audio + fn.random.normal(device=preprocessing_device) * dither_coeff
+
+    audio = fn.preemphasis_filter(audio, preemph_coeff=preemph_coeff)
+
+    spec = fn.spectrogram(
+        audio,
+        nfft=nfft,
+        window_length=window_size * sample_rate,
+        window_step=window_stride * sample_rate,
+    )
+
+    mel_spec = fn.mel_filter_bank(
+        spec, sample_rate=sample_rate, nfilter=nfeatures, normalize=True
+    )
+
+    log_features = fn.to_decibels(
+        mel_spec, multiplier=np.log(10), reference=1.0, cutoff_db=math.log(1e-20)
+    )
+
+    log_features_len = fn.shapes(log_features)
+    if frame_splicing_factor != 1:
+        log_features_len = _div_ceil(log_features_len, frame_splicing_factor)
+
+    log_features = fn.normalize(log_features, axes=[1])
+    log_features = fn.pad(log_features, axes=[1], fill_value=0, align=pad_align, shape=(-1,))
+
+    if train_pipeline and do_spectrogram_masking:
+        anchors, shapes = fn.external_source(
+            source=cutouts_generator, num_outputs=2, cycle=True
+        )
+        log_features = fn.erase(
+            log_features,
+            anchor=anchors,
+            shape=shapes,
+            axes=[0, 1],
+            fill_value=0,
+            normalized_anchor=True,
+        )
+
+    # When modifying DALI pipeline returns, make sure you update `output_map`
+    # in DALIGenericIterator invocation
+    if not is_triton_pipeline:
+        return log_features.gpu(), label.gpu(), log_features_len.gpu()
+    else:
+        return fn.cast(log_features.gpu(), dtype=types.FLOAT16)
+
+
+def serialize_dali_pipeline(filepath):
+    p = dali_asr_pipeline(
+        train_pipeline=False,
+        file_root=None,
+        file_list=None,
+        sample_rate=16000,
+        silence_threshold=-60,
+        resample_range=None,
+        discrete_resample_range=None,
+        window_size=.02,
+        window_stride=.01,
+        nfeatures=64,
+        nfft=512,
+        frame_splicing_factor=1,
+        dither_coeff=1e-5,
+        pad_align=16,
+        preemph_coeff=.97,
+        preprocessing_device="gpu",
+        is_triton_pipeline=True,
+        batch_size=1,
+        num_threads=1,
```

Comment:
Do we want only a single CPU thread for this?

Reply:
Good catch. I'll set more than one thread.